packages/pwa/app/commerce-api/hooks/useBasket.js

```diff
@@ -392,6 +392,26 @@ export default function useBasket(opts = {}) {
                 return api.shopperBaskets.getShippingMethodsForShipment({
                     parameters: {basketId: basket.basketId, shipmentId: 'me'}
                 })
+            },
+
+            /**
+             * Merge data from the previous shopper's basket into the current shopper's active basket
+             * and delete the previous shopper's basket.
+             */
+            async mergeBasket() {
+                const response = api.shopperBaskets.mergeBasket({
+                    options: {
+                        parameters: {
+                            createDestinationBasket: true // If the current shopper has an active basket, this parameter is ignored.
+                        }
+                    }
+                })
+
+                if (response.fault) {
+                    throw new Error(response)
+                }
+
+                setBasket(response)
```

Comment:
The fact that we are talking to two different APIs at the same time, we need to make sure the `response` are very similar if not have the exact same structure. otherwise we could run into issues.

Imagine a scenario that the OCAPI methods calls `setBasket(response)` and saves `{id: xxx, ...}` and SCAPI saves `{basketId:xxx , ...}` to the state, it could cause issues.

If the responses have similar format, then great! If not, we need to parse the response into a common interface.

Reply:
The basket response for both SCAPI and OCAPI are identical. We should be fine returning the response directly.

Reply:
awesome